src/core/shader.js

```diff
@@ -60,9 +60,7 @@ Shader.prototype = {
     var self = this;
     var variables = {};
     var schema = this.schema;
-    var squemaKeys = Object.keys(schema);
-    squemaKeys.forEach(processSquema);
-    function processSquema (key) {
```

Comment:
That indirection is good because usually you don't want to know the details. You just want to read first the function and get a general idea of what it does. If you want to know what processSquema does you can go and read it but otherwise is noise to get a general understanding.